File: solutions/dart/loops.dart
import 'dart:math';

void main(List<String> arguments) {
  var iterations = int.parse(arguments[0]);
  simulate(iterations);
}

void simulate(int iterations) {
  final start = new DateTime.now().millisecondsSinceEpoch;
  const sampleSize = 23;

  var count = 0;
  final random = new Random();
  for (var i=0; i < iterations; i++) {
    final arr = new List.filled(365, 0);
    for (var j=0; j < sampleSize; j++) {
      final number = random.nextInt(365);
      if (arr[number] == 1) {
        count++;
        break;
      } else {
        arr[number] = 1;
      }
    }
  }
  print('iterations: $iterations');
  print('sample-size: $sampleSize');
  final results = (count / iterations * 100).toStringAsFixed(2);
  print('percent: $results');
  final end = new DateTime.now().millisecondsSinceEpoch;
  final diff = ((end - start) / 1000).toStringAsFixed(3);
  print('seconds: $diff');
}

File: solutions/typescript/loops.ts
const iterations = parseInt(process.argv[2]);

function simulate() {
  const start = process.hrtime.bigint();
  const sampleSize = 23

  let count = 0
  for (let i = 0; i < iterations; i++) {
    const arr = new Array(365)
    for (let j = 0; j < sampleSize; j++) {
      const rand = Math.floor(Math.random() * 365)
      if (arr[rand]) {
        count++
        break
      } else {
        arr[rand] = 1
      }
    }
  }
  console.log(`iterations: ${iterations}`)
  console.log(`sample-size: ${sampleSize}`)
  const results = (count / iterations * 100).toFixed(2)
  console.log(`percent: ${results}`)
  const end = process.hrtime.bigint();
  const diff = (Number(end - start) / 1_000_000_000).toFixed(6)
  console.log(`seconds: ${diff}`)
}

simulate()
